test_tracetools/test/test_generic_pub_sub.py

```diff
@@ -0,0 +1,312 @@
+# Copyright 2021 Christophe Bedard
+# Copyright 2024 TIER IV, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from tracetools_test.case import TraceTestCase
+from tracetools_trace.tools import tracepoints as tp
+from tracetools_trace.tools.lttng import is_lttng_installed
+
+
+@unittest.skipIf(not is_lttng_installed(minimum_version='2.9.0'), 'LTTng is required')
+class TestGenericPubSub(TraceTestCase):
+
+    def __init__(self, *args) -> None:
+        super().__init__(
+            *args,
+            session_name_prefix='session-test-generic-pub-sub',
+            events_ros=[
+                tp.rmw_publisher_init,
+                tp.rcl_publisher_init,
+                tp.rmw_publish,
+                tp.rcl_publish,
+                tp.rclcpp_publish,
+                tp.rmw_subscription_init,
+                tp.rcl_subscription_init,
+                tp.rclcpp_subscription_init,
+                tp.rclcpp_subscription_callback_added,
+                tp.callback_start,
+                tp.callback_end,
```

Comment:
Is there any reason why the `rmw_take` tracepoint is missing from the test? The `rmw_*` PRs add an `rmw_take` tracepoint for serialized takes, so it should ideally be tested here.

Reply:
@christophebedard 
Sorry, I implemented `test_generic_*.py` with reference to it before https://github.com/ros2/ros2_tracing/commit/dadd0e9a1a9f4b8e224a0dbfd78b7f96e48bc6af.
I didn't notice that it had been updated.
I fixed `test_generic_pub_sub.py` in https://github.com/ros2/ros2_tracing/pull/97/commits/983cc237b50739e314a045da6c9c23886323d1de.
Also, `test_generic_subscription.py` was updated, so I fixed in https://github.com/ros2/ros2_tracing/pull/97/commits/9ddf39e1ac4497326ad5ff071b42feea352e7995.

Reply:
Thank you for updating the new tests!